reflect/src/main/java/dagger/reflect/Scope.java

```diff
@@ -198,7 +260,17 @@ Builder addBindingIntoMap(Key key, Object entryKey, Binding entryValueBinding) {
         throw new IllegalArgumentException("key.type() must be Map");
       }
 
-      bindings.addIntoMap(key, entryKey, entryValueBinding);
+      Map<Object, Binding> mapBinding = mapBindings.get(key);
+      //noinspection Java8MapApi Supporting old Android API levels.
+      if (mapBinding == null) {
+        mapBinding = new LinkedHashMap<>();
+        mapBindings.put(key, mapBinding);
+      }
+      Binding replaced = mapBinding.put(entryKey, entryValueBinding);
+      if (replaced != null) {
+        throw new IllegalStateException(); // TODO duplicate keys
```

Comment:
should this throw the same error as on line 183?

Reply:
Yeah it's close. The error message would have to detail that the problem is the `@MapKey` annotation being the problem. Exception messages have mostly been ignored since they can be cleaned up later and don't affect functionality.